Cython/Compiler/ExprNodes.py

```diff
@@ -208,10 +208,13 @@ def make_dedup_key(outer_type, item_nodes):
               type(node.constant_result) if node.type is py_object_type else None) if node.has_constant_result()
         else None  # something we cannot handle => short-circuit below
         for node in item_nodes
-    ]
+    )
     if None in item_keys:
         return None
-    return outer_type, tuple(item_keys)
+
+    if outer_type.name == 'frozenset':
```

Comment:
It's ok if you don't like the specific code suggestion that I gave you, but then please use something equivalent or better. This should use a builtin type check, not just a name check.

Reply:
Sorry, I missed the builtin type check. I added it